components/Reference/DynamicFee.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState, useContext } from 'react'
+
+import debounce from 'lodash.debounce'
+import { IOpcode } from 'types'
+
+import { EthereumContext } from 'context/ethereumContext'
+
+import { calculateDynamicFee } from 'util/gas'
+
+import { Input, Radio, Icon } from 'components/ui'
+import { H2 } from 'components/ui/Doc'
+
+const debounceTimeout = 100 // ms
+
+type Props = {
+  opcode: IOpcode
+}
+
+type InputValue = {
+  [name: string]: string
+}
+
+const DynamicFee = ({ opcode }: Props) => {
+  const { dynamicFee } = opcode
+
+  const { common } = useContext(EthereumContext)
+  const [inputs, setInputs] = useState<InputValue>({})
+  const [result, setResult] = useState('0')
+
+  // Initialize inputs with default keys & values
+  useEffect(() => {
+    const inputValues: InputValue = {}
+    Object.keys(dynamicFee?.inputs || []).map((key: string) => {
+      if (dynamicFee?.inputs[key].type === 'number') {
+        inputValues[key] = '0'
+      } else if (dynamicFee?.inputs[key].type === 'boolean') {
```

Comment:
I don't like boolean for warm or cold (from the user perspective), can we use radio buttons?

Reply:
We can style it as a toggle, but its still something that has two values only, on/off - so its a boolean in the opcode.json meta

Reply:
Yes in the json it's fine, would like to change how it's shown :-)
A button that toggles with written warm on side and cold the other is perfect, like @dorlevi suggested